frontend/src/component/common/Sticky/StickyProvider.tsx

```diff
@@ -0,0 +1,76 @@
+import { useState, useCallback, ReactNode, RefObject } from 'react';
+import { StickyContext } from './StickyContext';
+
+interface IStickyProviderProps {
+    children: ReactNode;
+}
+
+export const StickyProvider = ({ children }: IStickyProviderProps) => {
+    const [stickyItems, setStickyItems] = useState<RefObject<HTMLDivElement>[]>(
+        [],
+    );
+
+    const registerStickyItem = useCallback(
+        (item: RefObject<HTMLDivElement>) => {
+            setStickyItems((prevItems) => {
+                if (item && !prevItems.includes(item)) {
+                    const newItems = [...prevItems, item];
+                    return newItems.sort((a, b) => {
+                        const elementA = a.current;
+                        const elementB = b.current;
+                        if (elementA && elementB) {
+                            return (
+                                elementA.getBoundingClientRect().top -
+                                elementB.getBoundingClientRect().top
+                            );
+                        }
+                        return 0;
+                    });
+                }
+
+                return prevItems;
+            });
+        },
+        [],
+    );
+
+    const unregisterStickyItem = useCallback(
+        (ref: RefObject<HTMLDivElement>) => {
+            setStickyItems((prev) => prev.filter((item) => item !== ref));
+        },
+        [],
+    );
+
+    const getTopOffset = useCallback(
+        (ref: RefObject<HTMLDivElement>) => {
+            return stickyItems.some((item) => item === ref)
+                ? stickyItems
+                      .slice(
+                          0,
+                          stickyItems.findIndex((item) => item === ref),
+                      )
+                      .reduce((acc, item) => {
+                          return item === ref
+                              ? acc
+                              : acc +
```

Comment:
It's verify difficult to read with this formatting. Also without intention revealing names I have on clue what it's doing